pkg/util/interval/generic/doc.go

```diff
@@ -24,7 +24,12 @@ Users of the package should follow these steps:
     internal/contract.go.
  3. Include a go generate declaration that invokes the gen.sh script with the
     type name as the first argument and the package name as the second argument.
- 4. Invoke go generate.
+    This is done for documentation purposes and will require a corresponding entry
```

Comment:
TBH I'm not 100% certain why we do this; I'm just documenting the procedure as I understand it.

Reply:
The `go:generate` stanzas are generally "legacy" bits of code. Generally the `go:generate` pre-dated the existence of the Bazel code generation. There isn't an expectation that one be added assuming you are writing new code.